hudi-common/src/main/java/org/apache/hudi/common/table/timeline/LSMTimeline.java

```diff
@@ -169,15 +168,17 @@ public static int latestSnapshotVersion(HoodieTableMetaClient metaClient, Storag
    * Returns all the valid snapshot versions.
    */
   public static List<Integer> allSnapshotVersions(HoodieTableMetaClient metaClient, StoragePath archivePath) throws IOException {
-    if (!metaClient.getStorage().exists(archivePath)) {
```

Comment:
I think, as the long-term solution, it makes sense to remove the `exists` API to avoid such exists calls and follow the pattern of a try-catch approach for FS call efficiency.